app/src/main/java/com/cp/tasky/auth/shared/data/repository/AuthRepositoryImpl.kt

```diff
@@ -0,0 +1,33 @@
+package com.cp.tasky.auth.shared.data.repository
+
+import com.cp.tasky.auth.shared.domain.AuthRepository
+import com.cp.tasky.auth.login.data.remote.models.LoginApiBody
+import com.cp.tasky.auth.login.data.remote.models.LoginApiResponse
+import com.cp.tasky.auth.shared.domain.model.LoginResponse
+import com.cp.tasky.auth.shared.domain.model.UserCredential
+import kotlin.math.log
+
+class AuthRepositoryImpl(
+    private val remoteDataSource: RemoteDataSource
+): AuthRepository {
+
+    override suspend fun loginUser(userCredential: UserCredential): LoginResponse {
+       return remoteDataSource.loginUser(
+           LoginApiBody(
+               email = userCredential.email,
+               password = userCredential.password
+           )
+       ).toLoginResponse()
+    }
+
+    private fun LoginApiResponse.toLoginResponse(): LoginResponse {
+        return LoginResponse(
+            accessToken = accessToken,
+            refreshToken = refreshToken,
+            fullName = fullName,
+            userId = userId,
+            accessTokenExpirationTimestamp = accessTokenExpirationTimestamp,
+            message = message
+        )
+    }
```

Comment:
We typically put mappers in a separate file since it's a separate responsibility than the one from the repository (it calls the mappers, but doesn't define how mapping works)